src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -239,6 +240,17 @@ class Direct$Type$Buffer$RW$$BO$
 #end[rw]
     }
 
+    // Constrain the capacity to int range throwing IAE if not possible
+    //
+    private static int clampCapacity(long capacity) {
+        try {
+            return Math.toIntExact(capacity);
+        } catch (ArithmeticException ignore) {
+            throw new IllegalArgumentException("capacity " + capacity
```

Comment:
I think we should mention the source, that the capacity value originates from a native call to NewDirectByteBuffer, since it that will not appear in any frames of the repoted stack trace.

Reply:
> I think we should mention the source, that the capacity value originates from a native call to NewDirectByteBuffer [...]

So you mean in the IAE message itself?

Reply:
Yes, enhance the message with more detail